homeassistant/components/netatmo/light.py

```diff
@@ -0,0 +1,142 @@
+
+    async def get_entities():
+        """Retrieve Netatmo entities."""
+        await data_handler.register_data_class(data_class)
+
+        entities = []
+        try:
+            all_cameras = []
+            for home in data_handler.data[data_class].cameras.values():
+                for camera in home.values():
+                    all_cameras.append(camera)
+
+            for camera in all_cameras:
+                if camera["type"] == "NOC":
+                    if not data_handler.webhook:
+                        raise PlatformNotReady
+
+                    _LOGGER.debug(
+                        "Adding camera light %s %s", camera["id"], camera["name"]
+                    )
+                    entities.append(
+                        NetatmoLight(
+                            data_handler,
+                            data_class,
+                            camera["id"],
+                            camera["type"],
+                            camera["home_id"],
+                        )
+                    )
+
+        except pyatmo.NoDevice:
+            _LOGGER.debug("No cameras found")
+
+        await data_handler.unregister_data_class(data_class)
+        return entities
+
+    async_add_entities(await get_entities(), True)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Netatmo camera platform."""
+    return
+
+
+class NetatmoLight(LightEntity, NetatmoBase):
+    """Representation of a Netatmo Presence camera light."""
+
+    def __init__(
+        self, data_handler, data_class, camera_id: str, camera_type: str, home_id: str
+    ):
+        """Initialize a Netatmo Presence camera light."""
+        LightEntity.__init__(self)
+        NetatmoBase.__init__(self, data_handler)
+
+        self._data_classes.append({"name": data_class})
+        self._id = camera_id
+        self._home_id = home_id
+        self._model = camera_type
+        self._device_name = self._data.get_camera(camera_id).get("name")
+        self._name = f"{MANUFACTURER} {self._device_name}"
+        self._is_on = False
+        self._unique_id = f"{self._id}-light"
+
+    async def async_added_to_hass(self) -> None:
+        """Entity created."""
+        await NetatmoBase.async_added_to_hass(self)
+
+        async def handle_event(event):
+            """Handle webhook events."""
+            data = event.data["data"]
+
+            if not data.get("event_type"):
+                return
+
+            if not data.get("camera_id"):
+                return
+
+            if (
+                data["home_id"] == self._home_id
+                and data["camera_id"] == self._id
+                and data["push_type"] == "NOC-light_mode"
+            ):
+                if data["sub_type"] in ["off", "auto"]:
+                    self._is_on = False
+
+                elif data["sub_type"] == "on":
+                    self._is_on = True
+
+                self.schedule_update_ha_state()
```

Comment:
```suggestion
                self.async_write_ha_state()
```
Can we write state directly instead since we are already in async context?